File: benchmarks/context-base/agents/ts/__template.ts
// Starter template — copy this file to create a new benchmark agent.
import { startAgent, type ExecutableTool } from "../../scaffolding/ts/index.js";
import type { SetupBody, SendMessageBody, SendMessageResponse } from "../../lib/protocol.js";

export function createCallbacks() {
  let state: any = {};

  return {
    setup: async (tools: ExecutableTool[], config: SetupBody["config"]) => {
      state = {};
    },

    sendMessage: async (body: SendMessageBody): Promise<SendMessageResponse> => {
      if (!state) throw new Error("Agent not set up");

      return {
        content: "the result of your agent",
        toolCalls: [],
        usage: {
          totalTokens: 0,
          inputTokens: 0,
          outputTokens: 0,
          cachedInputTokens: 0,
          outputReasoningTokens: 0,
        },
        durationMs: 0,
        debug: {
          systemPrompt: "",
          toolNames: [],
          modelResponse: "",
          toolCallsMade: [],
        },
      };
    },
  };
}

import { fileURLToPath } from "node:url";
import { resolve } from "node:path";
if (fileURLToPath(import.meta.url) === resolve(process.argv[1])) {
  const cbs = createCallbacks();
  startAgent({ setup: cbs.setup, sendMessage: cbs.sendMessage });
}

File: benchmarks/context-base/agents/py/__template.py
"""Python baseline agent stub.

Returns hardcoded responses to verify the cross-language protocol works.
Replace with real LLM integration (e.g., openai pip package) for actual benchmarking.
"""

import sys
import os

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "..", "scaffolding", "py"))
from scaffolding import start_agent

tools = []
config = {}


def setup(tool_defs, cfg):
    global tools, config
    tools = tool_defs
    config = cfg


def send_message(body):
    return {
        "content": "(Python stub) No LLM integration yet.",
        "toolCalls": [],
        "usage": {"totalTokens": 0, "inputTokens": 0, "outputTokens": 0},
        "durationMs": 0,
    }


if __name__ == "__main__":
    start_agent(setup=setup, send_message=send_message)

